services/integration-content-repository/app/api/swagger/swagger.json

```diff
@@ -886,52 +140,83 @@
           }
         ],
         "tags": [
-          "edges"
+          "flows"
         ],
-        "summary": "Stores a edge with form data",
-        "operationId": "addEdge",
-        "parameters": [
-          {
-            "name": "oihid",
-            "in": "path",
-            "description": "ID of the flow the edge gets added to",
-            "required": true,
-            "schema": {
-              "type": "string",
-              "format": "string"
+        "summary": "Add a new flow to the repository",
+        "description": "The new Flow will automatically be associated with the current user through a \"user\"-type relationship, allowing editing and viewing by this user irrespective of tenancies.",
+        "operationId": "addFlow",
+        "responses": {
+          "200": {
+            "description": "The newly created Flow.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Flow"
+                }
+              }
             }
           },
-          {
-            "name": "edgeid",
-            "in": "path",
-            "description": "ID of the edge",
-            "required": true,
-            "schema": {
-              "type": "string",
-              "format": "string"
+          "409": {
+            "description": "Flow with this oihid already exists, oihid must be unique"
+          }
+        },
+        "requestBody": {
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/MutableFlow"
+              }
             }
+          },
+          "description": "A Flow object to be added to the repository.",
+          "required": true
+        }
+      },
+      "put": {
+        "x-swagger-router-controller": "flow",
+        "security": [
+          {
+            "Bearer": [
+
+            ]
           }
         ],
+        "tags": [
+          "flows"
+        ],
+        "summary": "Update an existing flow",
+        "description": "The oihid is used to match the existing Flow that will be updated. Note that missing fields in the input will overwrite existing fields with null. The current version and version history is updated automatically.",
+        "operationId": "updateFlow",
         "responses": {
           "200": {
-            "description": "The new edge",
+            "description": "The updated Flow.",
             "content": {
               "application/json": {
                 "schema": {
-                  "$ref": "#/components/schemas/EdgeResponse"
+                  "$ref": "#/components/schemas/Flow"
                 }
               }
             }
           },
-          "405": {
-            "description": "Invalid input"
+          "404": {
+            "description": "Flow not found"
           }
```

Comment:
We should also add code 201 here [API Service Guidelines](https://github.com/openintegrationhub/openintegrationhub/blob/icr_new_flow_model/Guidelines/ServiceAPIGuidelines.md#must-use-http-methods-correctly):

> successful PUT requests will usually generate 200 or 204 (if the resource was updated - with or without actual content returned), **and 201 (if the resource was created)**

Reply:
At the moment, we don't have upsert, just update and insert as two individual endpoints. 201 is only for created resources.